sdk/azidentity/azidentity_test.go

```diff
@@ -33,32 +40,27 @@ func Test_SetEnvAuthorityHost(t *testing.T) {
 	if authorityHost != envHostString {
 		t.Fatalf("Unexpected error when get host from environment vairable: %v", err)
 	}
-
-	// Unset that host environment vairable to avoid other tests failed.
-	err = os.Unsetenv("AZURE_AUTHORITY_HOST")
-	if err != nil {
-		t.Fatalf("Unexpected error when unset environment vairable: %v", err)
-	}
 }
 
 func Test_CustomAuthorityHost(t *testing.T) {
 	err := os.Setenv("AZURE_AUTHORITY_HOST", envHostString)
+	defer func() {
+		// Unset that host environment vairable to avoid other tests failed.
+		err = os.Unsetenv("AZURE_AUTHORITY_HOST")
+		if err != nil {
+			t.Fatalf("Unexpected error when unset environment vairable: %v", err)
+		}
+	}()
 	if err != nil {
 		t.Fatalf("Unexpected error when initializing environment variables: %v", err)
 	}
-
 	authorityHost, err := setAuthorityHost(customHostString)
 	if err != nil {
 		t.Fatal(err)
 	}
-	if authorityHost != customHostString {
-		t.Fatalf("Unexpected error when get host from environment vairable: %v", err)
-	}
-
-	// Unset that host environment vairable to avoid other tests failed.
-	err = os.Unsetenv("AZURE_AUTHORITY_HOST")
-	if err != nil {
-		t.Fatalf("Unexpected error when unset environment vairable: %v", err)
+	// env takes priority
```

Comment:
Why?

Reply:
It lets you change the value without having to crack the code.